crates/paimon/src/file_index/file_index_format.rs

```diff
@@ -0,0 +1,517 @@
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+
+use crate::{
+    io::{FileIO, FileRead, FileStatus, InputFile, OutputFile},
+    Error,
+};
+
+/// Default 1MB read block size
+const READ_BLOCK_SIZE: u64 = 1024 * 1024;
+
+/// Quoted from the Java implement of the structure,
+/// `MAGIC`` is used to mark the beginning of a FileFormat structure.
+pub const MAGIC: u64 = 1493475289347502;
+
+/// Used to mark an empty INDEX.
+pub const EMPTY_INDEX_FLAG: i64 = -1;
+
+#[derive(Debug)]
+struct IndexInfo {
+    start_pos: i64,
+    length: i64,
+}
+
+#[repr(i32)]
+#[derive(Debug, PartialEq, Eq)]
+enum Version {
+    V1,
+}
+
+/// File index file format. All columns and offsets are stored in the header.
+///
+/// ```text
+///   _____________________________________    _____________________
+/// ｜     magic    ｜version｜head length ｜
+/// ｜-------------------------------------｜
+/// ｜            column number            ｜
+/// ｜-------------------------------------｜
+/// ｜   column 1        ｜ index number   ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 1 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 2 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 3 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜            HEADER
+/// ｜   column 2        ｜ index number   ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 1 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 2 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜
+/// ｜  index name 3 ｜start pos ｜length  ｜
+/// ｜-------------------------------------｜
+/// ｜                 ...                 ｜
+/// ｜-------------------------------------｜
+/// ｜                 ...                 ｜
+/// ｜-------------------------------------｜
+/// ｜  redundant length ｜redundant bytes ｜
+/// ｜-------------------------------------｜    ---------------------
+/// ｜                BODY                 ｜
+/// ｜                BODY                 ｜
+/// ｜                BODY                 ｜             BODY
+/// ｜                BODY                 ｜
+/// ｜_____________________________________｜    _____________________
+///
+/// - `magic`: 8 bytes long
+/// - `version`: 4-byte integer
+/// - `head length`: 4-byte integer
+/// - `column number`: 4-byte integer
+/// - `column x`: variable-length UTF-8 string (length + bytes)
+/// - `index number`: 4-byte integer (number of index items below)
+/// - `index name x`: variable-length UTF-8 string
+/// - `start pos`: 4-byte integer
+/// - `length`: 4-byte integer
+/// - `redundant length`: 4-byte integer (for compatibility with future versions; content is zero in this version)
+/// - `redundant bytes`: variable-length bytes (for compatibility with future versions; empty in this version)
+/// - `BODY`: sequence of index data (concatenated index data for each column)
+/// ```
+pub async fn write_column_indexes(
```

Comment:
Add the link: https://github.com/apache/paimon/blob/release-0.8.2/paimon-common/src/main/java/org/apache/paimon/fileindex/FileIndexFormat.java ?

Reply:
ok